### PR TITLE
ares: add darwin support

### PR DESCRIPTION
**Pull request by wegank**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/emulators/bsnes/ares/default.nix [hunk 1]
```diff
@@ -31,45 +34,58 @@ stdenv.mkDerivation (finalAttrs: {
   patches = [
     ./000-dont-rebuild-on-install.patch
     ./001-fix-ruby.patch
+    ./002-sips-to-png2icns.patch
```
Comment (viraptor) [inline pkgs/applications/emulators/bsnes/ares/default.nix:37]: I read that sips has been included in sdk 10.3, so we should have it available in 11.0. Is it missing? If so, maybe it's an issue to raise.

Comment (wegank) [reply pkgs/applications/emulators/bsnes/ares/default.nix:37]: Yes. And the issue actually dates back to #102263.

Comment (wegank) [reply pkgs/applications/emulators/bsnes/ares/default.nix:37]: In the meantime, I guess sips comes with macOS itself, not with the SDK.

**Review by viraptor — APPROVED** [review]

File: pkgs/applications/emulators/bsnes/ares/default.nix [hunk 2]
```diff
@@ -15,6 +16,8 @@
 , libpulseaudio
 , openal
 , udev
+, Cocoa
+, OpenAL
```
Comment (AndersonTorres) [inline pkgs/applications/emulators/bsnes/ares/default.nix:20]: Given the issue #204303, 
```suggestion
, darwin
```

Comment (AndersonTorres) [inline pkgs/applications/emulators/bsnes/ares/default.nix:20]: Given the issue #204303, 
```suggestion
, darwin
```

File: pkgs/applications/emulators/bsnes/ares/default.nix [hunk 3]
```diff
@@ -15,6 +16,8 @@
...
 }:
```
Comment (AndersonTorres) [inline pkgs/applications/emulators/bsnes/ares/default.nix:21]:
```suggestion
}:
let
  inherit (darwin.apple_sdk_11_0.frameworks) Cocoa OpenAL;
in
```

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -2162,7 +2162,9 @@ with pkgs;
 
   ### APPLICATIONS/EMULATORS/BSNES
 
-  ares = callPackage ../applications/emulators/bsnes/ares { };
+  ares = darwin.apple_sdk_11_0.callPackage ../applications/emulators/bsnes/ares {
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:2165]: Is this hack still needed?

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -2162,7 +2162,9 @@ with pkgs;
...
+    inherit (darwin.apple_sdk_11_0.frameworks) Cocoa OpenAL;
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:2166]:
```suggestion
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/applications/emulators/bsnes/ares/default.nix (hunk 2 above)
Comment (wegank) [reply pkgs/applications/emulators/bsnes/ares/default.nix]: Done.

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (wegank) [reply pkgs/top-level/all-packages.nix:2165]: Even on aarch64-darwin, without setting `NIX_CFLAGS_COMPILE`, I had something like

```
make: *** [../nall/GNUmakefile:244: obj/ares-n64-rdp.o] Error 1
make: *** Waiting for unfinished jobs....
In file included from ../thirdparty/mame/mame/video/n64.cpp:28:
In file included from ../thirdparty/mame/mame/includes/n64.h:9:
../thirdparty/mame/mame/video/n64.h:132:7: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
class n64_rdp : public poly_manager<uint32_t, rdp_poly_state, 8>
      ^
../thirdparty/mame/mame/video/n64.cpp:3169:10: note: in implicit destructor for 'n64_rdp' first required here
n64_rdp::n64_rdp(n64_state &state, uint32_t* rdram, uint32_t* dmem) : poly_manager<uint32_t, rdp_poly_state, 8>(state.machine())
```

So I think it only builds on SDK 10.14+.

**Pull request merged**